### PR TITLE
feat(dynamodb): Support point-in-time recovery

### LINKED ISSUE #935: Support "point_in_time_recovery" for AWS DynamoDB
We need the "point_in_time_recovery" function for AWS DynamoDB.
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/PointInTimeRecovery.html

Implement point_in_time_recovery

### PR DESCRIPTION
**Pull request by atarax**
### Description of your changes

"Fixes #935 ": Introduced the option to specify "Point in time recovery" for dynamodb-tables

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Code has been tested on a kind-cluster created by running `make e2e` with skipping cleanup and running controller locally.

### CODE REVIEW

File: pkg/controller/dynamodb/table/hooks.go
Comment (MisterMX) [inline pkg/controller/dynamodb/table/hooks.go:133]: You can use [`k8s.io/utils/pointer`](https://pkg.go.dev/k8s.io/utils/pointer) to simplify this

```suggestion
		pitrSpecEnabled := pointer.BoolDeref(cr.Spec.ForProvider.PointInTimeRecoveryEnabled, false)
```

File: pkg/controller/dynamodb/table/hooks.go
Comment (MisterMX) [inline pkg/controller/dynamodb/table/hooks.go:107]: I don't see where this function is used.

**Review by MisterMX — CHANGES_REQUESTED** [review]
Thank you very much for your contribution, @atarax! Can you take a look at the remarks I made in the code?

File: pkg/controller/dynamodb/table/hooks.go
Comment (tkaesserfm) [reply pkg/controller/dynamodb/table/hooks.go:133]: nice, thank you :)

Comment (atarax) [reply pkg/controller/dynamodb/table/hooks.go]: nice, thank you! :)

File: pkg/controller/dynamodb/table/hooks.go
Comment (MisterMX) [inline pkg/controller/dynamodb/table/hooks.go:449]: This can be simplified

```suggestion
        return pointer.StringDeref(pitrStatus, "") == string(svcapitypes.PointInTimeRecoveryStatus_ENABLED)
```

**Review by MisterMX — CHANGES_REQUESTED** [review]
@atarax one last nitpick. Everything else looks good to me.

Can you also squash your changes into a single commit and sign it before merge?

File: pkg/controller/dynamodb/table/hooks.go
Comment (atarax) [reply pkg/controller/dynamodb/table/hooks.go:449]: Sure no worries :) Some maintainers like if it's split so i wasn't sure. Gonna apply this and squash. Thanks!

Comment (atarax) [reply pkg/controller/dynamodb/table/hooks.go]: nitpick applied and squashed

File: table.yaml
Comment (MisterMX) [inline table.yaml:1]: This file shouldn't be committed.

File: test
Comment (MisterMX) [inline test:1]: This file shouldn't be committed.

**Review by MisterMX — CHANGES_REQUESTED** [review]
@atarax sorry I forgot the two last files that are part of your local test setup and shouldn't be in this PR. Can you remove them?

File: table.yaml
Comment (atarax) [reply table.yaml:1]: sorry for that i squashed in a hurry

**Pull request merged**
